### DISCUSSION TITLE
Local server vs. remote/static CSS transition timing

### DISCUSSION THREAD

**Discussion by MrFussyfont** [vercel/next.js#23865] [Help]
My issue started with noticing inconsistent behavior in Chrome with my on-page-load CSS transitions (animations), but only in Chrome, and only after building the static site and deploying. I've tried several different ways of coding and triggering the transitions, but all of them break to some degree (e.g., only one of the transitions runs, the others appear to have finished early). My latest attempt, using React Transition Group, also breaks in Firefox, but again only on the remote server.

Desired behavior running locally (npm start):
![anim1b](https://user-images.githubusercontent.com/474540/114233993-227b4400-994c-11eb-927f-ae240215c341.gif)

Same code statically generated running remotely (npm run build, copied to server or deployed on Netlify):
![anim2b](https://user-images.githubusercontent.com/474540/114234191-6e2ded80-994c-11eb-8497-6cb4d10418b9.gif)

Code looks like this (orange circle is the header):
```
<CSSTransition in={true} appear={true} classNames="header" timeout={5000}>
  <header>
    <img... etc>

--- CSS ---
.header-appear { // this class added on render by React CSSTransition
  clip-path: circle(0%);
  transition: clip-path 2s;
}
.header-appear img {
  transform: rotate(180deg);
  transition: transform 2s 1s; // all the sub-transitions are delayed, e.g., img starts rotating one second after clip-path starts
}
.header-appear-active { // this class added in the next tick, triggering the transitions
  clip-path: circle(100%);
}
.header-appear-active img {
  transform: rotate(0);
}
```

An earlier version breaking on Chrome (also statically generated, on remote server):
![anim3b](https://user-images.githubusercontent.com/474540/114234401-afbe9880-994c-11eb-8594-9bfe3ff3c1ed.gif)

In that version, I used `useEffect()` with `setTimeout()` to add an "animate" class to the body, which triggers the same transitions. This works sometimes in Chrome, always in Firefox, and always locally in both browsers.

What would cause this difference between the Next local server version and the statically generated version? Is this a bug? Is there a workaround? It seems to have something to do with the execution timing or loading of the CSS and/or JS.

Next version  10.0.9  
React 17.0.1 
Problem observed in both macOS and Windows.

**Comment by vinhocent** [comment]
Did you ever find a solution to this?

**Reply by vinhocent** [reply]
I just fixed this issue for me - I'm not sure what's behind the solution. But my animations started working consistently on Chrome when I changed my component from:

```
export default function Home() {
    return( ... ) // your html here
}
```
to:

```
const Home: NextPage = () => {
   return( ... ) // your html here
}
export default Home
```

**Reply by MrFussyfont** [reply]
Thanks, I don't think I did, but IIRC got around the problem by adding a security prompt before the animation starts, something I had to do anyway. But I'll keep this in mind the next time I want to try something like this.
